src/Persistence/Mapping/Driver/MappingDriverChain.php

```diff
@@ -73,7 +73,7 @@ public function getDrivers()
     public function loadMetadataForClass(string $className, ClassMetadata $metadata)
     {
         foreach ($this->drivers as $namespace => $driver) {
-            if (strpos($className, $namespace) === 0) {
+            if (strpos($className, $namespace . '\\') === 0) {
```

Comment:
The `$namespace` can already contain a `\` at the end. It'd be better to ensure there's always a single `\` at the end in the `addDriver` method